satellite/audit/service.go

```diff
@@ -58,13 +61,64 @@ func NewService(log *zap.Logger, config Config, metainfo *metainfo.Service,
 	}, nil
 }
 
+// ReservoirService is a temp name for the service struct during the audit 2.0 refactor.
+// Once V3-2363 and V3-2364 are implemented, Service2 will replace the existing Service struct.
+type ReservoirService struct {
+	log *zap.Logger
+
+	reservoirSlots int
+	Reservoirs     map[storj.NodeID]*Reservoir
+	rand           *rand.Rand
+
+	MetainfoLoop  *metainfo.Loop
+	ReservoirLoop sync2.Cycle
+}
+
+// NewReservoirService instantiates Service2
+func NewReservoirService(log *zap.Logger, metaLoop *metainfo.Loop, config Config) (*ReservoirService, error) {
+	r := rand.New(rand.NewSource(time.Now().Unix()))
+	return &ReservoirService{
+		log: log,
+
+		reservoirSlots: config.Slots,
+		Reservoirs:     make(map[storj.NodeID]*Reservoir),
+		rand:           r,
+
+		MetainfoLoop:  metaLoop,
+		ReservoirLoop: *sync2.NewCycle(config.Interval),
+	}, nil
+}
+
+// Run runs auditing service 2.0
+func (service *ReservoirService) Run(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	service.log.Info("audit 2.0 is starting up")
+
+	return service.ReservoirLoop.Run(ctx, func(ctx context.Context) (err error) {
+		defer mon.Task()(&ctx)(&err)
+		pathCollector := NewPathCollector(service.reservoirSlots, service.rand)
+		err = service.MetainfoLoop.Join(ctx, pathCollector)
+		if err != nil {
+			service.log.Error("error joining metainfoloop", zap.Error(err))
+			return nil
+		}
+		// We iterate over path collector reservoirs to deep copy them because map is a reference type.
+		service.Reservoirs = make(map[storj.NodeID]*Reservoir, len(pathCollector.Reservoirs))
```

Comment:
maybe instead `service.Reservoirs = pathCollector.Reservoirs`

Reply:
We iterate over path collector reservoirs to deep copy them because map is a reference type.

Reply:
I know it's a reference type.

Reply:
sorry- deleted comment after I saw your other comment explaining why lol